### ISSUE TITLE
Access code-server from coder response 502 Gad Gateway

### ISSUE THREAD

**Issue by lvchenyang** [coder/coder#12790]
502 - Bad Gateway
Failed to proxy request to application: proxyconnect tcp: dial context: connect tcp [fd7a:115c:a1e0:42ba:af8f:1cde:1c68:a816]:7890: connection was refused

<img width="940" alt="image" src="https://github.com/coder/coder/assets/5507872/659f9601-070b-473a-a7a6-15101a7e955a">

```

+ mkdir -p /tmp/code-server/lib /tmp/code-server/bin
+ tar -C /tmp/code-server/lib -xzf ~/.cache/code-server/code-server-4.19.1-linux-arm64.tar.gz
+ mv -f /tmp/code-server/lib/code-server-4.19.1-linux-arm64 /tmp/code-server/lib/code-server-4.19.1
+ ln -fs /tmp/code-server/lib/code-server-4.19.1/bin/code-server /tmp/code-server/bin/code-server
Standalone release has been installed into /tmp/code-server/lib/code-server-4.19.1
Extend your path to use code-server:
  PATH="/tmp/code-server/bin:$PATH"
Then run with:
  code-server
Deploy code-server for your team with Coder: https://github.com/coder/coder
```

**Comment by matifali** [comment]
Are you sure `code-server` is running?

**Comment by bfqrst** [comment]
I think this hit us too! For some additional context, it seems to have appeared in v2.8.0 and it's still around in the latest v2.9.x. The v2.7 versions are not affected by this. Running v2.7.3 at the moment.

**Comment by jacobwi** [comment]
Same issue after the update

**Comment by bfqrst** [comment]
![image](https://github.com/coder/coder/assets/7373518/b2dc427f-8867-408d-8599-7028e53a477a)

So a couple of things we noticed:

1. The IPv6 stuff is similar in all screenshots and seems to be related to the [Tailscale](https://tailscale.com/kb/1201/4via6-subnets) networking stuff, so that is where I would focus the hunt
2. The port in our screenshot is 3128 which is our corporate proxy port. That seems to get picked up along the way. Unsure if 7890 is also some proxy in case of @lvchenyang 
3. We made sure that code-server comes up correctly with our configured port and is curl-able from inside the machine

Hope that helps!
Cheers

**Comment by spikecurtis** [comment]
@coadler this could be related to https://github.com/coder/coder/pull/11810

@lvchenyang @bfqrst @jacobwi is `code-server` listening port bound to IPv4, IPv6 or both?

**Comment by bfqrst** [comment]
So this is the command that starts code-server: `/usr/lib/code-server/lib/node /usr/lib/code-server --auth none --port 8080`. Not sure that if you don't explicitly configure something it's both stacks... In terms of netstat, code-server only listens to IPv4 as far as I can tell:

```
netstat -tulpn 
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:8787            0.0.0.0:*               LISTEN      2277/rserver        
tcp        0      0 0.0.0.0:111             0.0.0.0:*               LISTEN      -                   
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      -                   
tcp        0      0 127.0.0.1:6060          0.0.0.0:*               LISTEN      1823/./coder        
tcp        0      0 127.0.0.1:8080          0.0.0.0:*               LISTEN      2349/node           
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      -                   
tcp        0      0 127.0.0.1:2112          0.0.0.0:*               LISTEN      1823/./coder        
tcp        0      0 127.0.0.1:2113          0.0.0.0:*               LISTEN      1823/./coder        
tcp6       0      0 :::111                  :::*                    LISTEN      -                   
tcp6       0      0 :::22                   :::*                    LISTEN      -                   
udp        0      0 127.0.0.53:53           0.0.0.0:*                           -                   
udp        0      0 10.194.73.0:68          0.0.0.0:*                           -                   
udp        0      0 0.0.0.0:111             0.0.0.0:*                           -                   
udp        0      0 127.0.0.1:323           0.0.0.0:*                           -                   
udp        0      0 0.0.0.0:41581           0.0.0.0:*                           1823/./coder        
udp6       0      0 :::40399                :::*                                1823/./coder        
udp6       0      0 :::111                  :::*                                -                   
udp6       0      0 ::1:323                 :::*                                -
```

Was that your question @spikecurtis?

**Comment by matifali** [comment]
I am getting into this too on v2.9.0+ while using [vscode-web](https://registry.coder.com/modules/vscode-web) module. Wildcard domains are correctly configured.

**Comment by coadler** [comment]
@bfqrst does the port shown in the error screen say `[ipv6 address]:8080`? Could you also confirm the url in your `coder_app` is `http://localhost:8080`?

**Comment by bfqrst** [comment]
> ![image](https://private-user-images.githubusercontent.com/7373518/318816417-b2dc427f-8867-408d-8599-7028e53a477a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIyNTc4NTAsIm5iZiI6MTcxMjI1NzU1MCwicGF0aCI6Ii83MzczNTE4LzMxODgxNjQxNy1iMmRjNDI3Zi04ODY3LTQwOGQtODU5OS03MDI4ZTUzYTQ3N2EucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDRUMTkwNTUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTA3NDVhYjU4ZWQyZGYyZWNkMjdmN2RhMDFlMDc5YTk1ODYzNTJkMzVkMzE2M2FmYzQzNTBlZDNkZWRlOWZkNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.PydSfiAVouY_9LUVOpYsNQf7hGI3_uWqer-LAue-oG4)
> 
> So a couple of things we noticed:
> 
> 1. The IPv6 stuff is similar in all screenshots and seems to be related to the [Tailscale](https://tailscale.com/kb/1201/4via6-subnets) networking stuff, so that is where I would focus the hunt
> 2. The port in our screenshot is 3128 which is our corporate proxy port. That seems to get picked up along the way. Unsure if 7890 is also some proxy in case of @lvchenyang
> 3. We made sure that code-server comes up correctly with our configured port and is curl-able from inside the machine
> 
> Hope that helps! Cheers

The above is our error screenshot. It shows the IPv6 address in brackets, but the port is our corporate proxy port `3128`. So that somehow enters the mix.


And yes, code-server runs under: `url          = "http://localhost:8080/?folder=/home/${local.linux_user}"`.

**Comment by coadler** [comment]
Sorry, completely missed that, thanks for the info. Last question, is coder being accessed over the proxy port you mentioned? i.e., is it somewhere in the browser URL?

**Comment by bfqrst** [comment]
No worries, thanks for looking into it!

The `3128` HAS to be the port we're setting as our HTTP[S]_PROXY and NO_PROXY vars on the respective workspaces/machines. In other words from the machine to the outside world. The other way around is: browser --> ingress (Traefik) --> Coder --> workspace from template (AWS EC2 based)

Does that make sense?

**Comment by coadler** [comment]
That makes sense. So is the `HTTP[S]_PROXY` env also set in the `coder` pods themselves as well?

**Comment by bfqrst** [comment]
> That makes sense. So is the `HTTP[S]_PROXY` env also set in the `coder` pods themselves as well?

In our case it's [Nomad](https://www.nomadproject.io/) jobs not pods, but your assessment seems to be true nonetheless!

Thanks @coadler

**Comment by coadler** [comment]
This is fixed now and should be included in the next `2.10.x` release
